httphandlers/http_handlers.go

```diff
@@ -238,18 +243,16 @@ func getUUIDsFromURI(uri string) []string {
 	return re.FindAllString(uri, -1)
 }
 
-func getRequestHeaders(req *http.Request, additionalFilter []string) map[string]string {
+func getRequestHeaders(req *http.Request, filter HeaderFilterCallback) map[string]string {
 
 	allowed := func(key string) bool {
 		for _, r := range headerDenyList {
 			if r.MatchString(key) {
 				return false
 			}
 		}
-		for _, f := range additionalFilter {
-			if strings.EqualFold(f, key) {
-				return false
-			}
+		if filter != nil {
+			return !filter(key)
```

Comment:
If the idea of `filter` is to provide the user of the library to provide custom logic for filtering, we can save ourselves the negation here. I think it also makes the code harder to reason about since the result of the `allowed` function below is negated again.